crates/flipperzero/src/lib.rs

```diff
@@ -16,3 +17,56 @@ pub mod __internal {
     // Re-export for use in macros
     pub use ufmt;
 }
+
+// Infrastructure for running unit tests.
```

Comment:
This infrastructure will likely end up becoming a proc macro (so it can be used for both unit tests and integration tests), but for now I'm implementing it explicitly to make its actions clear.

Reply:
I've now made it a proc macro, because it ended up being easier (as then I could use a second proc macro to process the tests themselves into a form that the test runner can use).